pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -0,0 +1,575 @@
+    return const_cast<Result>(&GLOBAL_RESULT_FALSE_CONST);
+}
+
+auto LightningKokkosSimulator::One() const -> Result {
+    return const_cast<Result>(&GLOBAL_RESULT_TRUE_CONST);
+}
+
+void LightningKokkosSimulator::NamedOperation(
+    const std::string &name, const std::vector<double> &params,
+    const std::vector<QubitIdType> &wires, bool inverse,
+    const std::vector<QubitIdType> &controlled_wires,
+    const std::vector<bool> &controlled_values) {
+    RT_FAIL_IF(!controlled_wires.empty() || !controlled_values.empty(),
+               "LightningKokkos does not support native quantum control.");
+
+    // Check the validity of number of qubits and parameters
+    RT_FAIL_IF(!isValidQubits(wires), "Given wires do not refer to qubits");
+    RT_FAIL_IF(!isValidQubits(controlled_wires),
+               "Given controlled wires do not refer to qubits");
+
+    // Convert wires to device wires
+    auto &&dev_wires = getDeviceWires(wires);
+
+    // Update the state-vector
+    this->device_sv->applyOperation(name, dev_wires, inverse, params);
+
+    // Update tape caching if required
+    if (this->tape_recording) {
+        this->cache_manager.addOperation(name, params, dev_wires, inverse, {},
+                                         {/*controlled_wires*/},
+                                         {/*controlled_values*/});
+    }
+}
+
+void LightningKokkosSimulator::MatrixOperation(
+    const std::vector<std::complex<double>> &matrix,
+    const std::vector<QubitIdType> &wires, bool inverse,
+    const std::vector<QubitIdType> &controlled_wires,
+    const std::vector<bool> &controlled_values) {
+    using UnmanagedComplexHostView =
+        Kokkos::View<Kokkos::complex<double> *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+    // TODO: Remove when controlled wires API is supported
+    RT_FAIL_IF(
+        !controlled_wires.empty() || !controlled_values.empty(),
+        "LightningKokkos device does not support native quantum control.");
+    RT_FAIL_IF(!isValidQubits(wires), "Given wires do not refer to qubits");
+    RT_FAIL_IF(!isValidQubits(controlled_wires),
+               "Given controlled wires do not refer to qubits");
+
+    // Convert wires to device wires
+    auto &&dev_wires = getDeviceWires(wires);
+
+    std::vector<Kokkos::complex<double>> matrix_kok;
+    matrix_kok.resize(matrix.size());
+    std::transform(
+        matrix.begin(), matrix.end(), matrix_kok.begin(),
+        [](auto c) { return static_cast<Kokkos::complex<double>>(c); });
+
+    Kokkos::View<Kokkos::complex<double> *> gate_matrix("gate_matrix",
+                                                        matrix_kok.size());
+    Kokkos::deep_copy(gate_matrix, UnmanagedComplexHostView(matrix_kok.data(),
+                                                            matrix_kok.size()));
+
+    // Update the state-vector
+    this->device_sv->applyMultiQubitOp(gate_matrix, dev_wires, inverse);
```

Comment:
Can't we let the base device handle this? 
```suggestion
    this->device_sv->applyMatrix(reinterpret_cast<Kokkos::complex<double> *>(matrix.data()), dev_wires, inverse);
```